Code/CIPLabeler/Wrap/pyCIPLabelsValidation.py

```diff
@@ -0,0 +1,138 @@
+# coding: utf-8
+import os
+import sys
+from rdkit import Chem
+
+EXPECTED_LABELS_OVERRIDES = {
+  # p/m labels are filtered out on input, so they are not
+  # included here
+
+  # allene-likes
+  'VS063': ['26R', '2S'],
+  'VS118': [],
+  'VS135': [],
+  'VS154': ['6E', '7E'],
+  'VS164': ['21R'],
+  'VS188': [],
+  'VS189': [],
+  'VS190': [],
+  'VS214': [],
+  'VS231': [],
+
+  # Chiralities not flagged by RDKit
+  'VS132': [],
+  'VS207': ['10r', '2s', '5s', '7r'],
+  'VS226': ['12r', '15r', '19s', '22s', '28s', '2r', '31s', '5r'],
+  'VS246': [],
+  'VS247': [],
+  'VS248': [],
+
+  # This one is worrying: the missing chirality '2R'
+  # causes '7S', '15S' to flip into '7r', '15s'
+  'VS279': ['12s', '15s', '4r', '7r'],
+
+  # Chiralities not flagged by RDKit (Rule 6)
+  'VS280': [],
+  'VS281': [],
+  'VS282': [],
+  'VS283': [],
+  'VS284': [],
+  'VS285': [],
+  'VS286': [],
+  'VS288': [],
+  'VS289': [],
+  'VS290': [],
+  'VS291': [],
+  'VS292': [],
+  'VS293': ['11Z', '12Z', '3Z', '4Z', '7Z', '8Z'],
+  'VS294': ['10S', '2S', '6R', '9R'],
+  'VS295': ['10R', '2S', '6R', '9S'],
+  'VS296': [],
+  'VS297': [],
+  'VS298': [],
+  'VS299': [],
+  'VS300': [],
+}
+
+
+def split_label_string(labels):
+  """
+    Splits the string of expected labels and filters out helical
+    labels, which are not supported by rdkit
+    """
+  return {label for label in labels.split() if label[-1] not in 'mpMP'}
+
+
+def supplier(fname):
+  """
+    Read the smiles, name and expected labels from the input file.
+    If we know of an override for the expected labels, we do the
+    override here.
+    """
+  with open(fname) as f:
+    for line in f:
+      smiles, name, expected, _ = line.split('\t', 3)
+
+      try:
+        expected = set(EXPECTED_LABELS_OVERRIDES[name])
+      except KeyError:
+        expected = split_label_string(expected)
+
+      mol = Chem.MolFromSmiles(smiles)
```

Comment:
You really don't want to be sanitizing here. If you sanitize in `MolFromSmiles()` it also calls `AssignStereochemistry()`, which uses the current (wrong) stereochemistry code to identify whether or not an atom can be a stereocenter.
It's better to do:
```
        mol = Chem.MolFromSmiles(smiles, sanitize=False)
        Chem.SanitizeMol(mol)
        Chem.SetBondStereoFromDirections(mol)
```
That clears up all the expected failures except:
```
EXPECTED_LABELS_OVERRIDES = {
  # p/m labels are filtered out on input, so they are not
  # included here

  # allene-likes
  'VS063': ['26R', '2S'],
  'VS118': [],
  'VS135': [],
  'VS154': ['6E', '7E'],
  'VS164': ['21R'],
  'VS231': [],

  # Chiralities not flagged by RDKit
  'VS132': [],
}
```
The allenes are to be expected, but the problem with VS132 is a bug in the sanitization code, which shouldn't be removing the chiral flag on the two Ns in that case.

Reply:
Making these changes converts this into a better test of the changes here instead of being a holistic check of the RDKit's full chirality handling

Reply:
Wow, great!

On the down side, this makes the validation test slower (from 3.5 to 7 s on my machine), but I guess that's acceptable.